### PR TITLE
Add separation environment

### PR DESCRIPTION
**Pull request by ethax-ross**
### Context

We want to add a new `separation` environment alongside the existing `sandbox` environment that can be used by providers to test a version of ECF with all the NPQ functionality removed/disabled.

### Changes proposed in this pull request

- Add separation environment

This PR covers:

- Adding the `separation` environment (based off `sandbox`)
- Adding `Makefile` support
- Deploy on merging to `main`
- Seed data
- Enable time traveller for testing
- Add Rails credentials for new env

### Guidance to review

I have also setup a new BigQuery events dataset for this environment and copied the `sandbox` secrets to the separation keyvault.

[Link to separation env](https://cpd-ecf-separation-web.teacherservices.cloud/)

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -41,6 +41,10 @@ sandbox:
   <<: *default
   url: <%= ENV["DATABASE_URL"] %>
 
+separation:
+  <<: *default
+  database: "early_careers_framework_separation"
```
Comment (saliceti) [inline config/database.yml:46]: Is this required? The url contains the database created by terraform

Comment (ethax-ross) [reply config/database.yml:46]: Ah no that should be `url: <%= ENV["DATABASE_URL"] %>` I'll update 👍

**Review by saliceti — APPROVED** [review]

**Comment by johnake** [comment]
[create environment separation for early careers framework](https://trello.com/c/krgAXOqp/1929-create-environment-separation-for-early-careers-framework)

File: config/database.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ review:
 
 performance:
   <<: *default
-  database: 'early_careers_framework_performance'
```
Comment (cwrw) [inline config/database.yml:33]: not sure about updating this as its a special performance env, just wanted to double check?

Comment (ethax-ross) [reply config/database.yml]: yeah you're right, it doesn't look like its deployed to Azure; I'll revert this

**Review by cwrw — APPROVED** [review]

**Review by mooktakim — APPROVED** [review]

**Pull request merged**